assets/src/scripts/components/hydrograph/schema.js

```diff
@@ -0,0 +1,130 @@
+const { normalize: normalizr, schema } = require('normalizr');
+
+const { replaceHtmlEntities } = require('../../utils');
+
+
+// sourceInfo schema
+const siteCode = new schema.Entity('siteCodes', {}, {idAttribute: 'value'});
+const timeZone = new schema.Entity('timeZones', {}, {idAttribute: 'zoneAbbreviation'});
+const timeZoneInfo = new schema.Entity('timeZoneInfo', {
+    defaultTimeZone: timeZone,
+    daylightSavingsTimeZone: timeZone
+}, {idAttribute: value => `${value.daylightSavingsTimeZone.zoneAbbreviation}:${value.defaultTimeZone.zoneAbbreviation}:${value.siteUsesDaylightSavingsTime}`});
+const sourceInfo = new schema.Entity('sourceInfo', {
+    siteCode: [siteCode],
+    timeZoneInfo: timeZoneInfo
+}, {idAttribute: value => value.siteCode.map(s => s.value).join(':')});
+
+// variable schema
+//const variableCode = new schema.Entity('variableCodes', {}, {idAttribute: 'value'});
+const option = new schema.Entity('options', {}, {idAttribute: 'optionCode'});
+const variable = new schema.Entity('variables', {
+    //variableCode: variableCode,
+    options: [option]
+}, {
+    idAttribute: 'oid',
+    processStrategy: (variable) => {
+        // Eliminate unnecessary nesting on options and variableCode attributes
+        return {
+            ...variable,
+            options: variable.options.option,
+            variableName: replaceHtmlEntities(variable.variableName),
+            variableCode: variable.variableCode[0]
+        };
+    }
+});
+
+// timeSeries schema
+const qualifier = new schema.Entity('qualifiers', {}, {idAttribute: 'qualifierCode'});
+const method = new schema.Entity('methods', {}, {idAttribute: 'methodID'});
+const timeSeries = tsKey => new schema.Entity('timeSeries', {
```

Comment:
Style issue: I don't think I like using arrow functions when defining a function object. Is there a reason you do so?

Reply:
Just conciseness. But looking now, those should all perhaps be memoized..

Reply:
That's a great idea!